File: 1394 find-lucky-integer-in-an-array/solution.go
func findLucky(arr []int) int {
    c:=make([]int,550)
    for _,v:= range arr {
        c[v]++
    }
    m:=-1
    for _,v:=range arr {
        if v==c[v] {
            m = max(m,v)
        }
    }
    return m
}

File: 1394 find-lucky-integer-in-an-array/solution.rs
impl Solution {
    pub fn find_lucky(arr: Vec<i32>) -> i32 {
        let mut c = vec![0;550];
        for &i in &arr {
            c[i as usize]+=1;
        }
        let mut m = -1;
        for &i in & arr {
            if i==c[i as usize] {
                m = m.max(i);
            }
        }
        m
    }
}
